servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/single/CallableSingleTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api.single;
+
+import io.servicetalk.concurrent.SingleSource;
+import io.servicetalk.concurrent.api.Single;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.Callable;
+
+import static io.servicetalk.concurrent.api.SourceAdapters.toSource;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class CallableSingleTest {
+
+    private Callable<Integer> factory;
+
+    @SuppressWarnings("unchecked")
+    @Before
+    public void setUp() throws Exception {
+        factory = mock(Callable.class);
+    }
+
+    @Test
+    public void testEverySubscribeCalls() throws Exception {
```

Comment:
Can you also add a test to verify the case when `Subscriber#onSubscribe()` throws an `Exception`?

Reply:
You can use the following:

```java
@Test
public void onSubscribeThrows() {
    final Single<Integer> source = Single.fromCallable(factory);
    @SuppressWarnings("unchecked")
    final SingleSource.Subscriber<Integer> subscriber = mock(SingleSource.Subscriber.class);
    doThrow(DELIBERATE_EXCEPTION).when(subscriber).onSubscribe(any());
    toSource(source).subscribe(subscriber);
    verify(subscriber).onSubscribe(any());
    verify(subscriber).onError(DELIBERATE_EXCEPTION);
    verifyNoMoreInteractions(subscriber);
}
```

Also you can create the `Subscriber` mock in the constructor just like the factory.